kuwala/core/canvas/src/pages/DataSourcePreview.js

```diff
@@ -1,238 +1,103 @@
-            </div>
-        )
-    }
-
-    const tableSelectionOnlick = async (addressString) => {
-        setSelectedTable(addressString)
-        setIsTableDataPreviewLoading(true)
-        const params = generateParamsByDataSourceType(selectedSource.data_catalog_item_id, addressString)
-        const res = await getTablePreview({
-            id: dataIndex,
-            params
-        });
-
-        if(res.status === 200) {
-            let cols = res.data.columns.map((el,i)=>{
-                return {
-                    Header: el,
-                    accessor: el,
-                }
-            });
+        const selectedAddress = selectedTable.split('@');
 
-            cols = [{
-                Header: "#",
-                id: "row",
-                filterable: false,
-                width: 50,
-                Cell: (row) => {
-                    return <div>{row.index+1}</div>;
-                }
-            }, ...cols]
-
-            setTableDataPreview({
-                columns: cols,
-                rows: getDataDictionary(res.data.rows, res.data.columns),
-            });
+        const columnsArray = tableDataPreview.columns.slice(1).map((el) => `${el.Header}`);
+        const payload = {
+            data_source_id: selectedSource.id,
+            name: `${selectedSource.dataCatalogItemId}_${selectedAddress[2]}`,
+            columns: columnsArray,
         }
-        setIsTableDataPreviewLoading(false)
-    }
 
-    const getDataDictionary = (data, headers) => {
-        let dictionary = [];
-        data.map((row,i) => {
-            let obj = {};
-            row.map((cell, j) => {
-                obj[headers[j]] = typeof data[i][j] === 'object' ? JSON.stringify(data[i][j]) : data[i][j];
-            })
-            dictionary.push(obj)
-        })
-        return dictionary;
-    };
-
-    const generateParamsByDataSourceType = (type, addressString) => {
-        const arr = addressString.split('@')
-        switch (type){
-            case "postgres":
-                return {
-                    schema_name: arr[0],
-                    table_name: arr[2],
-                    limit_columns: 200,
-                    limit_rows: 300,
-                }
-            case "bigquery":
-                return {
-                    project_name: arr[0],
-                    dataset_name: arr[1],
-                    table_name: arr[2],
-                    limit_columns: 200,
-                    limit_rows: 300,
-                }
-            default: return ""
+        switch (selectedSource.dataCatalogItemId) {
+            case("bigquery"):
+                payload.dataset_name = selectedAddress[1]
+                payload.table_name = selectedAddress[2]
+                break;
+            case("postgres"):
+                payload.schema_name = selectedAddress[0]
+                payload.table_name = selectedAddress[2]
```

Comment:
Please use property instead of index access as the order of the properties isn't ensured (see comment above).

Reply:
This one should be okay, since the array is splitted from an exact string so the array arrangement shouldn’t change. below is an example.

![image](https://user-images.githubusercontent.com/12572145/161793109-9898f0dd-4278-4582-96a0-528953eabf34.png)